modules/engage-theodul-plugin-tab-downloads/src/main/resources/static/main.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2009-2011 The Regents of the University of California Licensed
+ * under the Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.osedu.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+/*jslint browser: true, nomen: true*/
+/*global define*/
+define(["jquery", "underscore", "backbone", "engage/core"], function($, _, Backbone, Engage) {
+    "use strict";
+
+    var insertIntoDOM = true;
+    var PLUGIN_NAME = "Downloads";
+    var PLUGIN_TYPE = "engage_tab";
+    var PLUGIN_VERSION = "1.0";
+    var PLUGIN_TEMPLATE_DESKTOP = "templates/desktop.html";
+    var PLUGIN_TEMPLATE_MOBILE = "templates/mobile.html";
+    var PLUGIN_TEMPLATE_EMBED = "templates/embed.html";
+    var PLUGIN_STYLES_DESKTOP = [
+        "styles/desktop.css"
+    ];
+    var PLUGIN_STYLES_EMBED = [
+        "styles/embed.css"
+    ];
+    var PLUGIN_STYLES_MOBILE = [
+        "styles/mobile.css"
+    ];
+
+    var plugin;
+    var events = {
+        plugin_load_done: new Engage.Event("Core:plugin_load_done", "", "handler"),
+        mediaPackageModelError: new Engage.Event("MhConnection:mediaPackageModelError", "", "handler")
+    };
+
+    var isDesktopMode = false;
+    var isEmbedMode = false;
+    var isMobileMode = false;
+
+    // desktop, embed and mobile logic
+    switch (Engage.model.get("mode")) {
+        case "embed":
+            plugin = {
+                insertIntoDOM: insertIntoDOM,
+                name: PLUGIN_NAME,
+                type: PLUGIN_TYPE,
+                version: PLUGIN_VERSION,
+                styles: PLUGIN_STYLES_EMBED,
+                template: PLUGIN_TEMPLATE_EMBED,
+                events: events
+            };
+            isEmbedMode = true;
+            break;
+        case "mobile":
+            plugin = {
+                insertIntoDOM: insertIntoDOM,
+                name: PLUGIN_NAME,
+                type: PLUGIN_TYPE,
+                version: PLUGIN_VERSION,
+                styles: PLUGIN_STYLES_MOBILE,
+                template: PLUGIN_TEMPLATE_MOBILE,
+                events: events
+            };
+            isMobileMode = true;
+            break;
```

Comment:
From what I can tell, these modes do the same thing: nothing.
Why handle them separately?